Add a docker-compose.yml and Makefile for running a mailserver

jakubgs opened the pull request:
This should simplify running a mailserver for a less tech-savvy user. All you need to do is run:
```
cd _assets/compose/mailserver && make
```
In the repo directory.

Also adds a [README](https://github.com/status-im/status-go/tree/add-mailserver-compose/_assets/compose/mailserver).

jakubgs committed:
add a docker-compose.yml and Makefile for running a mailserver

Signed-off-by: Jakub Sokołowski <jakub@status.im>

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [x] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?

mandrigin reviewed (approved):
awesome!

corpetty commented:
if attempting to run from a github clone (assuming you don't use go get properly, you get the following errors:

```
Current dir is /home/petty/GitHub/status/status-go, which seems to be different from your GOPATH.
Please, build status-go from GOPATH for proper build.
  GOPATH       = /home/petty/go
  Current dir  = /home/petty/GitHub/status/status-go
  Expected dir = /home/petty/go/src/github.com/status-im/status-go)
See https://golang.org/doc/code.html#GOPATH for more info
.  Stop.
```

jakubgs commented:
@corpetty ah yes, that's a Go thing. There is a check in that [Makefile at line 18](https://github.com/status-im/status-go/blob/develop/Makefile#L18):
```Makefile
EXPECTED_PATH=$(shell go env GOPATH)/src/github.com/status-im/status-go
ifneq ($(CURDIR),$(EXPECTED_PATH))
define NOT_IN_GOPATH_ERROR

Current dir is $(CURDIR), which seems to be different from your GOPATH.
Please, build status-go from GOPATH for proper build.
  GOPATH       = $(shell go env GOPATH)
  Current dir  = $(CURDIR)
  Expected dir = $(EXPECTED_PATH))
See https://golang.org/doc/code.html#GOPATH for more info

endef
$(error $(NOT_IN_GOPATH_ERROR))
endif
```
Which means every time - even if the target doesn't require it - this check is run and fails if repo was cloned to the wrong dir. Maybe it would be better for users if the instructions didn't say to run `make run-mailserver` but rather to do `cd _assets/compose/mailserver` and then `make`, to avoid this.

gravityblast commented:
> Maybe it would be better for users if the instructions didn't say to run `make run-mailserver` but rather to do `cd _assets/compose/mailserver` and then `make`, to avoid this.

yeah I think that would be easier for the user and more clear for the main project Makefile

corpetty commented:
I also seem to be getting an error with the `statusd` command that is run from the compose file.  `docker logs` only gives `statusd` usage prompts, not a running mailserver.

jakubgs commented:
Okay, changed the `config.json` target that generates it to just `config` so it's re-generated every time.
I also added checks for availability of `jq` and `docker-compose`:
```Makefile
checks:
ifeq (, $(shell which docker-compose))
	$(error No $(BOLD)docker-compoase$(RESET) in your $$PATH. Please install it)
endif
ifeq (, $(shell which jq))
	$(error No $(BOLD)jq$(RESET) in your $$PATH. Please install it)
endif
```

adambabik commented on _assets/compose/mailserver/Makefile:
```diff
@@ -0,0 +1,65 @@
+ifeq (, $(shell which docker-compose))
+	$(error No $(BOLD)docker-compoase$(RESET) in your $$PATH. Please install it)
+endif
+ifeq (, $(shell which jq))
```
Isn't it going to return an error when running `make checks` because of `jq` used in `BOOT_NODES`?

jakubgs replied:
You're right, I forgot to remove that part since we do not update config anyway.

adambabik commented on _assets/compose/mailserver/Makefile:
```diff
@@ -0,0 +1,65 @@
+	docker-compose logs -f -t --tail=100
+
+enode:
+	@docker-compose logs --tail=all | grep -oP '.*self="\Kenode://[^?]+' | sed 's/127.0.0.1/$(PUBLIC_IP)/'
```
Another option might be to use IPC file and `admin.nodeInfo` method but it assumes IPC is created.

jakubgs replied:
I thought this was simpler, but I guess I can try doing it with RPC.

jakubgs replied:
Done.

pedropombeiro commented on _assets/compose/mailserver/Makefile:
```diff
@@ -0,0 +1,65 @@
+
+checks:
+ifeq (, $(shell which docker-compose))
+	$(error No $(BOLD)docker-compoase$(RESET) in your $$PATH. Please install it)
```
```suggestion
	$(error No $(BOLD)docker-compose$(RESET) in your $$PATH. Please install it)
```

jakubgs replied:
Fixed.
